saleor/graphql/invoice/mutations.py

```diff
@@ -0,0 +1,273 @@
+import graphene
+from django.core.exceptions import ValidationError
+
+from ...core import JobStatus
+from ...core.permissions import OrderPermissions
+from ...graphql.invoice.enums import PendingTarget
+from ...invoice import events, models
+from ...invoice.emails import send_invoice
+from ...invoice.error_codes import InvoiceErrorCode
+from ...order import OrderStatus
+from ..core.mutations import ModelDeleteMutation, ModelMutation
+from ..core.types.common import InvoiceError
+from ..invoice.types import Invoice
+from ..order.types import Order
+
+
+class RequestInvoice(ModelMutation):
+    class Meta:
+        description = "Request an invoice for the order."
+        model = models.Invoice
+        permissions = (OrderPermissions.MANAGE_ORDERS,)
+        error_type_class = InvoiceError
+        error_type_field = "invoice_errors"
+
+    class Arguments:
+        order_id = graphene.ID(
+            required=True, description="ID of the order related to invoice."
+        )
+        number = graphene.String(
+            required=False, description="Invoice number (optional)."
```

Comment:
It would be useful to explain it a bit more e.g. "Invoice number (optional). Provide it if you want to set a specific number on the invoice. Otherwise, a unique number will be generated." - something like that?